rust/ql/lib/codeql/rust/dataflow/internal/DataFlowImpl.qll

```diff
@@ -1806,3 +1857,22 @@ private module Cached {
 }
 
 import Cached
+
+cached
+private module OptionalSteps {
+  cached
+  predicate optionalStep(Node node1, string name, Node node2) {
+    FlowSummaryImpl::Private::Steps::summaryReadStep(node1.(Node::FlowSummaryNode).getSummaryNode(),
+      TOptionalStep(name), node2.(Node::FlowSummaryNode).getSummaryNode()) or
+    FlowSummaryImpl::Private::Steps::summaryStoreStep(node1.(Node::FlowSummaryNode).getSummaryNode(),
+      TOptionalStep(name), node2.(Node::FlowSummaryNode).getSummaryNode())
```

Comment:
```suggestion
    FlowSummaryImpl::Private::Steps::summaryReadStep(node1.(Node::FlowSummaryNode).getSummaryNode(),
      TOptionalStep(name), node2.(Node::FlowSummaryNode).getSummaryNode())
```
With the previous suggestion, the store step should not be necessary.

Reply:
Yes that works, see also: https://github.com/github/codeql/pull/18960/commits/2804c130279cb74e4c5eee7851535ed998f15e00